File: references/cellab_devkit-1.0 (1)/ruledefs/vonpar.js
    /*

        3 bit parity rule for von Neumann neighbourhood implemented
        with the VONN3 own-code evaluator.

    */

    rule.worldtype = 13;         // 2D torus world
    rule.patreq = "square";
    rule.ocodereq = "vonn3";

    function vonpar(oldstate) {
        return (oldstate & 7) ^
               ((oldstate >> 3) & 7) ^
               ((oldstate >> 6) & 7) ^
               ((oldstate >> 9) & 7);
    }

File: references/cellab_devkit-1.0 (1)/rules_java/vonpar.java
/*

    3 bit parity rule for von Neumann neighbourhood implemented
    with the VONN3 own-code evaluator.

*/

class Vonpar extends ruletable {

    void jcruleModes() {
        setWorld(13);
        setPatternRequest("square");
        setOwnCodeRequest("vonn3");
    }

    int jcrule(int oldstate) {
        return BITFIELD(0, 3) ^ BITFIELD(3, 3) ^ BITFIELD(6, 3) ^ BITFIELD(9, 3);
    }
}

public class vonpar {
    public static void main(String args[]) {
        (new Vonpar()).generateRuleFile("vonpar");
    }
}
